File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            if target - nums[i] not in dict:
                dict[nums[i]] = i
            else:
                return [i, dict[target-nums[i]]]
        return [0]

# @lc code=end

File: 1.two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map<int,int> ma;
        for(int i=0; i<n; i++){
            ma[nums[i]] = i;
        }
        for(int i=0; i<n; i++){
            if(nums[i]+nums[ma[nums[i]]]==target && i!=ma[nums[i]]){
                return {i, ma[nums[i]]};
            }
        }
        return {};
    }
};